python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,240 @@
+        self._set_model_mixins()
+
+    def _set_revng_executable(self, revng_executable_path: str | None):
+        """
+        Check if the path of the user supplied or default revng executable
+        is present on the system
+        """
+        if not revng_executable_path:
+            revng_executable = shutil.which("revng")
+        else:
+            revng_executable = revng_executable_path
+        assert revng_executable is not None
+        assert os.access(revng_executable, os.F_OK | os.X_OK)
+        self.revng_executable = revng_executable
+
+    def _set_resume_dir(self, resume_dir: str | None):
+        """
+        Use the user supplied resume dir or create a temp dir
+        """
+        if not resume_dir:
+            self._tmp_dir = TemporaryDirectory()
+            resume_dir = self._tmp_dir.name
+        assert os.path.isdir(resume_dir)
+        self.resume_dir = resume_dir
+
+    def _set_model(self, model: Binary):
+        self.model = model
+        self.last_saved_model = deepcopy(self.model)
+        # We call this method in multiple places. During the initial auto
+        # analysis for example we don't have the pipeline description yet
+        # so we need to take that into account.
+        if self.pipeline_description:
+            self._set_model_mixins()
+
+    def _set_model_mixins(self):
+        for rank in self.pipeline_description["Ranks"]:
+            model_path = rank.get("TupleTreePath")
+            if not model_path:
+                continue
+
+            # Extract only the first path of the path, eg:
+            # /TypeDefinitions/$1/Entries/$2 ->
+            # /TypeDefinitions
+            if len(model_path) > 1:
+                model_path = f"/{model_path.split("/")[1]}"  # fmt: skip
+
+            obj = get_element_by_path(model_path, self.model)
+            if isinstance(obj, TypedList):
+                for elem in obj:
+                    elem._project = self
+            else:
+                obj._project = self
+
+    def _get_artifact_kind(self, step_name: str) -> str:
+        for step in self.pipeline_description["Steps"]:
+            if step["Name"] == step_name:
+                return step["Artifacts"]["Kind"]
+        raise RuntimeError(f"Couldn't find step: {step_name}")
+
+    def _get_artifact_container(self, step_name) -> str:
+        for step in self.pipeline_description["Steps"]:
+            if step["Name"] == step_name:
+                return step["Artifacts"]["Container"]
+        raise RuntimeError(f"Couldn't find step: {step_name}")
+
+    def _get_step_name(self, analysis_name: str) -> str:
+        for step in self.pipeline_description["Steps"]:
+            for analysis in step["Analyses"]:
+                if analysis["Name"] == analysis_name:
+                    return step["Name"]
+        raise RuntimeError(f"Couldn't find step for analysis: {analysis_name}")
+
+    def _get_result_mime(self, artifact_name: str, result: str | bytes) -> bytes | str:
```

Comment:
This function does not return the result mime. It seems to decode the result according to the mime.
Rename.

Reply:
Renamed to `_decode_result_on_mime`